library/src/main/java/com/novoda/storagepathfinder/DeviceStoragePath.java

```diff
@@ -0,0 +1,68 @@
+package com.novoda.storagepathfinder;
+
+import java.io.File;
+
+/**
+ * Wrapper around a string for a Storage path with a type.
+ */
+class DeviceStoragePath implements StoragePath {
+
+    public static StoragePath create(String path, DeviceStoragePath.Type type) {
+        return new DeviceStoragePath(path, type);
+    }
+
+    private final String path;
+    private final Type type;
+
+    private DeviceStoragePath(String path, Type type) {
+        this.path = path;
+        this.type = type;
+    }
+
+    @Override
+    public Type getType() {
+        return type;
+    }
+
+    @Override
+    public String asString() {
+        return path;
+    }
+
+    @Override
+    public File asFile() {
```

Comment:
`pathAsFile`.

Reply:
It is a path?  `path.pathAsFile()` doesn't make sense to me. `path.asFile()` reads more clearly

Reply:
`getPathAsFile()` and `getPathAsString()` make sense. 

I know you are not a fan of `getX` but its common Java convention so for a OS library I'd suggest we use it